src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/crafting/HopperListener.java

```diff
@@ -0,0 +1,35 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners.crafting;
+
+import javax.annotation.Nonnull;
+
+import io.github.thebusybiscuit.slimefun4.api.inventory.AbstractVanillaBlockInventory;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryMoveItemEvent;
+import org.bukkit.event.inventory.InventoryType;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+
+/**
+ * This {@link Listener} prevents item from being transferred to a
+ * {@link AbstractVanillaBlockInventory} using a hopper.
+ *
+ * @author CURVX
+ *
+ */
+public class HopperListener implements SlimefunCraftingListener {
```

Comment:
Why does this implement the SlimefunCraftingListener...?
It doesn't have anything to do with crafting...?